helix-term/src/ui/completion.rs

```diff
@@ -164,6 +164,21 @@ impl Completion {
                             );
                             doc.apply(&transaction, view.id);
                         }
+                    } else {
+                        let additional_text_edits =
+                            Completion::resolve_completion_item(doc, item.clone())
+                                .and_then(|item| item.additional_text_edits);
+
+                        if let Some(additional_edits) = additional_text_edits {
```

Comment:
That is a lot of code duplication! But I was not able to solve avoid that.
I wanted to do something like:
```rust
let additional_text_edits = item.additional_text_edits
    .or_else(|| Completion::resolve_completion_item(doc, item.clone()).and_then(|item| item.additional_text_edits))
```

But as `item` is a shared reference, I receive an error telling that can not move out "item.additional_text_edits" since "item" is a shared reference :(

I would appreciate any help with this one 😄

Reply:
Because you need to create a new Completion item when `&item.additional_text_edits` is `None`

I've get an idea like this (involve intruduce `text_edits_ref` reference, and `additional_text_edits` to catch the (maybe) new completion item:
```rust
let additional_text_edits;
let mut text_edits_ref = &item.additional_text_edits;
if text_edits_ref.is_none() {
    additional_text_edits =
        Completion::resolve_completion_item(doc, item.clone())
            .and_then(|item| item.additional_text_edits);
    text_edits_ref = &additional_text_edits;
}
if let Some(additional_edits) = text_edits_ref {
    ....
}
```

Reply:
Let's just do this:

```rust
if item.additional_text_edits.is_none() {
    item = Completion::resolve_completion_item(doc, item.clone()); // handle err here
});

if let Some(additional_edits) = &item.additional_text_edits {
```

Reply:
Thank you for your suggestions. I had a problem with them though, with both solutions the compiler complained saying that the "completion item" returned from `resolve_completion_item` does not live enough.

Anyway, I took a little different approach using your ideas, and reduced the code duplication at least, take a look at it and say me what you think